File: tests/output/angular/src/appgenerated/model/restaurant/reservation-item.ts
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

import {DO_BOOLEEN, DO_DATE_HEURE, DO_ID, DO_QUANTITE} from "@/domains";
import {e, entity, EntityToType} from "@focus4/entities";

export type ReservationItem = EntityToType<ReservationItemEntityType>;
export type ReservationItemEntityType = typeof ReservationItemEntity;

export const ReservationItemEntity = entity({
    id: e.field(DO_ID, f => f.optional()
        .label("restaurant.reservation.id")
    ),
    dateReservation: e.field(DO_DATE_HEURE, f => f
        .label("restaurant.reservation.dateReservation")
    ),
    nombrePersonnes: e.field(DO_QUANTITE, f => f
        .label("restaurant.reservation.nombrePersonnes")
    ),
    confirmee: e.field(DO_BOOLEEN, f => f.defaultValue(false)
        .label("restaurant.reservation.confirmee")
    ),
    clientId: e.field(DO_ID, f => f
        .label("restaurant.reservation.clientId")
    ),
    restaurantId: e.field(DO_ID, f => f
        .label("restaurant.reservation.restaurantId")
    )
});

File: tests/output/jpa/src/main/javagen/restaurant/jpa_identity_enums/dtos/restaurant/ReservationItem.java
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

package restaurant.jpa_identity_enums.dtos.restaurant;

import java.time.LocalDateTime;

import jakarta.annotation.Generated;

/**
 * Détail d'une réservation en liste.
 */
@Generated("TopModel : https://github.com/klee-contrib/topmodel")
public interface ReservationItem {

	/**
	 * Identifiant de la réservation.
	 */
	Integer getId();

	/**
	 * Date et heure de la réservation.
	 */
	LocalDateTime getDateReservation();

	/**
	 * Nombre de personnes.
	 */
	Integer getNombrePersonnes();

	/**
	 * Indique si la réservation est confirmée.
	 */
	Boolean getConfirmee();

	/**
	 * Client ayant fait la réservation.
	 */
	Integer getClientId();

	/**
	 * Restaurant concerné par la réservation.
	 */
	Integer getRestaurantId();

	/**
	 * Hydrate values of instance.
	 * @param id value to set.
	 * @param dateReservation value to set.
	 * @param nombrePersonnes value to set.
	 * @param confirmee value to set.
	 * @param clientId value to set.
	 * @param restaurantId value to set.
	 */
	void hydrate(Integer id, LocalDateTime dateReservation, Integer nombrePersonnes, Boolean confirmee, Integer clientId, Integer restaurantId);
}
